### PR TITLE
Implement count based on size

### LINKED ISSUE #2085: Function for the total number of items in an array/matrix
Here's overview how other libraries call their functions:

| Library | element count | each dimension |
| --- | --- | --- |
| numpy | [`size`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.size.html) | [`shape`](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.shape.html) |
| Julia | [`length`](https://docs.julialang.org/en/v1/base/arrays/#Base.length-Tuple{AbstractArray}) | [`size`](https://docs.julialang.org/en/v1/base/arrays/#Base.size) |
| mathjs | ? | `size` |

I'm not sure how to call it then... Do you have a good idea?

PHP calls its array length `count`. But using `numberOfElements` or `elementCount` seems a bit long.

_Originally posted by @Josef37 in https://github.com/josdejong/mathjs/pull/2075#discussion_r559786530_

### PR DESCRIPTION
**Pull request by Josef37**
Closes #2085

### CODE REVIEW

File: test/unit-tests/function/matrix/count.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    assert.strictEqual(count('123456'), 6)
+    assert.strictEqual(count(''), 0)
+  }
+
+  it('should count something with a size as Matrix', function () {
+    assertCount(math.count)
+  })
+
+  it('should count something with a size as Array', function () {
+    const math2 = math.create({ matrix: 'Array' })
```
Comment (josdejong) [inline test/unit-tests/function/matrix/count.test.js:22]: This doesn't work as you expect: the `matrix` config only applies to matrices created when parsing an expression. 

Can you create some straightforward units instead which explicitly create a DenseMatrix and SparseMatrix? Like

```
assert.strictEqual(count(matrix([[1, 2, 3], [4, 5, 6]])), 6)

assert.strictEqual(count(sparse([[1, 2, 3], [4, 5, 6]])), 6)

```

**Review by josdejong — COMMENTED** [review]
Thanks @Josef37 , looks good 👍 I made two comments about simplifying the code a bit more, and improving upon the unit tests

File: src/function/matrix/count.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   * @param {string | Array | Matrix} x  A matrix or string
+   * @return {number} An integer with the elements in `x`.
+   */
+  return typed(name, {
+    string: function (x) {
+      return x.length
+    },
+
+    'Matrix | Array': function (x) {
+      const dimensions = size(x)
```
[the remark this answers is not in the document]
Comment (Josef37) [reply src/function/matrix/count.js:34]: Seems to work. I was unsure about a possibly empty `size(x)` array. But it always returns at least `[0]` for Matrix or Array input.

[the remark this answers is not in the document]
Comment (Josef37) [reply src/function/matrix/count.js:34]: I could do the same implementation for strings. Would you prefer the uniformity or the performance?

File: test/unit-tests/function/matrix/count.test.js (hunk 1 above)
Comment (Josef37) [reply test/unit-tests/function/matrix/count.test.js:22]: Oh... I was confused about the setting in `size()`. But it's only applied in the `string` case there

File: src/function/matrix/count.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (josdejong) [reply src/function/matrix/count.js]: I think it's perfect like this 👍 . String is quite a different thing, and making function this _too_ generic can also result in weird cases, so it's good to keep the behavior for the different types explicit.

File: test/unit-tests/function/matrix/count.test.js (hunk 1 above)
Comment (josdejong) [reply test/unit-tests/function/matrix/count.test.js]: 👍

**Comment by josdejong** [comment]
Nice! Thanks for your latest improvement Josef! Will merge your PR now.

BTW not related to this speicfic PR, but I don't get why we stiff have `continuous-integration/travis-ci` listed in the checks, travis should be fully removed from the project (replaced with github workflow).

**Comment by Josef37** [comment]
I've never done much on the DevOps side of things. I'm kind of lost about that 😅

**Comment by josdejong** [comment]
yes I understand, it's just that I'm triggered by it with every PR 😄

**Pull request merged**